### DISCUSSION TITLE
Implementing passphrase caching with PGPainless

### DISCUSSION THREAD

**Discussion by bezmi** [zeapo/Android-Password-Store#2213] [Ideas]
Hi @msfjarvis, long time user of the app and I have been keenly following the v2.0 release. I have been reading through the crypto code to see how caching of passphrases might be achieved. Have you looked into this at all? From my reading into PGPainless, using the `CachingSecretKeyringProtector` would be a start?

**Comment by msfjarvis** [comment]
It's not really useful for us since we need to preserve the passphrase outside the scope of individual screens and the key protector does not live long enough for it. It'll have to be done like [this](https://github.com/open-keychain/open-keychain/blob/master/OpenKeychain/src/main/java/org/sufficientlysecure/keychain/service/PassphraseCacheService.java)

**Reply by bezmi** [reply]
okay, I saw this linked in the tracking issue, I'll take a look over it.

What is the purpose of the `addDecryptionPassphrase` when decrypting? From the PGPainless docs, it seems to be used for symmetrically encrypted messages.
